### PR TITLE
Doc updates and polishing

### PR DESCRIPTION
**Pull request by webframp**
Just basic polishing of docs. Fixing and adding a few links, adding alt text where missing, minor grammar and spelling changes.

Release notes section reformatted and reorganized.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -16,12 +18,14 @@
 - Document view of your outlines, for reading document-style Org files
   with long body text
 - Powerful support for linking between Org files
-- Org-mode integration
+- OrgMode integration with [org-mobile.el](http://orgmode.org/manual/MobileOrg.html)
```
Comment (mgmart) [inline index.md:21]: I checked Org's website and found the following names used synonymously:
- Org
- Org mode

in older documents the name is 'Org-mode'. I found no occurrence of 'OrgMode', though.

Comment (mgmart) [reply index.md:21]: I do like the simple 'Org'

File: index.md [hunk 2]
```diff
@@ -552,78 +547,101 @@ MobileOrg is thanks to the following:
...
-  - FEATURE: Support for iOS 9 & 10 (previous versions are not
-    supported any more - v.1.6.1 should be still available from the AppStore)
-  - UI: Old Org-mode icon is back
-  - UI: New icons	
-  - UI: Screens are adapted to new iOS L&F
+### MobileOrg 1.7.1
+
+#### Upgrades
+  - FEATURE: Support for iOS 9 & 10 
+    - Older versions are no longer supported. Sorry, its time I deal with the upgrade path.
```
Comment (mgmart) [inline index.md:563]: s/I/we ?

File: index.md [hunk 3]
```diff
@@ -552,78 +547,101 @@ MobileOrg is thanks to the following:
...
 
-- MobileOrg 1.7.1
-  - FEATURE: Support for iOS 9 & 10 (previous versions are not
-    supported any more - v.1.6.1 should be still available from the AppStore)
-  - UI: Old Org-mode icon is back
-  - UI: New icons	
-  - UI: Screens are adapted to new iOS L&F
+### MobileOrg 1.7.1
+
+#### Upgrades
```
Comment (mgmart) [inline index.md:561]: I like the new structure.

**Comment by webframp** [comment]
We can switch to just Org or Org mode everywhere. That's good to me and mirrors emacs usage anyway

**Pull request merged**